### PR TITLE
Demo Site: Add Redis Cache Handler

### PR DESCRIPTION
**Pull request by nsams**
## Description

This adds redis to docker-compose and a cache-handler implementation based on "a big project of vivid-planet".

I wanted to have the cache-handler written in typescript, but was not able to use ts-node for that. next.config.js needs the path (usually using require.resolve) to the cache-handler file, not the import. But when running using ts-node the js file doesn't exist, and ts is not supported. I didn't find a way to get this running with ts-node, so I switched that to a build-step using tsc also in dev.

Alternative: write cache-handler not in typescript (all examples I found do that)

## Open TODOs/questions

-   [x] Merge parent PR #2595
-   [ ] decide if the compile step (in dev) is Ok, or find a better solution
-   [ ] add to starter

### CODE REVIEW

File: demo/site/tsconfig.server.json [hunk 1]
```diff
@@ -8,5 +8,5 @@
         "isolatedModules": false,
         "noEmit": false
     },
-    "include": ["server.ts", "tracing.ts"]
+    "include": ["server.ts", "tracing.ts", "cache-handler.ts"]
```
Comment (johnnyomair) [inline demo/site/tsconfig.server.json:11]: The cache handler isn't really part of the server. But we can neglect this in favor of simplicity, I guess.

**Comment by johnnyomair** [comment]
> decide if the compile step (in dev) is Ok, or find a better solution

I think it's okay, why wouldn't it be?

Alternative: `@ts-check` comment, as we do for next.config.js: https://github.com/vivid-planet/comet-starter/blob/main/site/next.config.js#L3

**Comment by nsams** [comment]
> why wouldn't it be?

because it is not using ts-node anymore and requires the compile step also locally

> @ts-check

I would like to write "real" typescript

**Comment by johnnyomair** [comment]
> because it is not using ts-node anymore and requires the compile step also locally

IMO this doesn't make much difference.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**